planner/core/find_best_task.go

```diff
@@ -336,6 +336,9 @@ func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candida
 			candidates = append(candidates, ds.getIndexMergeCandidate(path))
 			continue
 		}
+		if ds.indexMergeHints != nil {
+			continue
```

Comment:
We cannot prune all other paths here if we have no index merge path, otherwise, we cannot generate a valid plan. For SQL hint, if it is inapplicable, we should just raise warning, instead of generating no plan.

Reply:
That means if there is no plan, we need to choose the table scan or a index scan and throw a warning?

Reply:
Yes